sktime/forecasting/prophet.py

```diff
@@ -0,0 +1,341 @@
+            stan_backend=stan_backend,
+            **kwargs
+        )
+
+    def fit(self, y, X=None, fh=None, **fit_params):
+        """Fit to training data.
+        Parameters
+        ----------
+        y : pd.Series
+            Target time series to which to fit the forecaster.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables.
+        fh : int, list or np.array, optional (default=None)
+            The forecasters horizon with the steps ahead to to predict.
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        y, X = check_y_X(y, X, index_type=pd.DatetimeIndex)
+        self._set_y_X(y, X)
+        self._set_fh(fh)
+
+        # We have to bring the data into the required format for fbprophet:
+        df = pd.DataFrame(y.rename("y"))
+        df["ds"] = y.index
+
+        # Add seasonality
+        if self.add_seasonality:
+            self._forecaster.add_seasonality(**self.add_seasonality)
+
+        # Add country holidays
+        if self.add_country_holidays:
+            self._forecaster.add_country_holidays(**self.add_country_holidays)
+
+        # Add regressor (multivariate)
+        if X is not None:
+            df = df.merge(X, left_index=True, right_on=X.index)
+            for col in X.columns:
+                self._forecaster.add_regressor(col)
+
+        self._forecaster.fit(df=df, **fit_params)
+        self._is_fitted = True
+        return self
+
+    def _predict(self, fh, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
+        """Predict
+
+        Parameters
+        ----------
+        fh : array-like
+            The forecasters horizon with the steps ahead to to predict.
+            Default is
+            one-step ahead forecast, i.e. np.array([1]).
+        X : pd.DataFrame, optional
+            Exogenous data, by default None
+        return_pred_int : bool, optional
+            Returns a pd.DataFrame with confidence intervalls, by default False
+        alpha : float, optional
+            Alpha level for confidence intervalls, by default DEFAULT_ALPHA
+
+        Returns
+        -------
+        y_pred : pandas.Series
+            Returns series of predicted values.
+
+        Raises
+        ------
+        Exception
+            Error when merging data
+        """
+        if type(fh._values) == pd.DatetimeIndex:
+            df = pd.DataFrame()
+            df["ds"] = fh._values
+        else:
+            # Try to create pd.DatetimeIndex
+            try:
+                periods = fh.to_pandas().max()
+                periods = fh.to_pandas().max()
+                df = self._forecaster.make_future_dataframe(
+                    periods=periods + 1, freq=self.freq, include_history=False)
+                df = df.iloc[fh.to_pandas()]
+            except Exception:
+                raise TypeError("Type of fh values must be int, np.array, list or pd.DatetimeIndex")
```

Comment:
I'd wrap this into a helper function - makes it more readable - `_coerce_to_datetime_index()` perhaps?

Reply:
done